src/Microsoft.Extensions.RazorViews.Sources/BaseView.cs

```diff
@@ -0,0 +1,311 @@
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> instance to write to.</param>
+        /// <param name="value">The <see cref="object"/> to write.</param>
+        /// <remarks>
+        /// <see cref="HelperResult.WriteTo(TextWriter)"/> is invoked for <see cref="HelperResult"/> types.
+        /// For all other types, the encoded result of <see cref="object.ToString"/> is written to the
+        /// <paramref name="writer"/>.
+        /// </remarks>
+        protected void WriteTo(TextWriter writer, object value)
+        {
+            if (value != null)
+            {
+                var helperResult = value as HelperResult;
+                if (helperResult != null)
+                {
+                    helperResult.WriteTo(writer);
+                }
+                else
+                {
+                    WriteTo(writer, Convert.ToString(value, CultureInfo.InvariantCulture));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> with HTML encoding to <paramref name="writer"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> instance to write to.</param>
+        /// <param name="value">The <see cref="string"/> to write.</param>
+        protected void WriteTo(TextWriter writer, string value)
+        {
+            WriteLiteralTo(writer, HtmlEncoder.Encode(value));
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> without HTML encoding to the <paramref name="writer"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> instance to write to.</param>
+        /// <param name="value">The <see cref="object"/> to write.</param>
+        protected void WriteLiteralTo(TextWriter writer, object value)
+        {
+            WriteLiteralTo(writer, Convert.ToString(value, CultureInfo.InvariantCulture));
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> without HTML encoding to <see cref="Output"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> instance to write to.</param>
+        /// <param name="value">The <see cref="string"/> to write.</param>
+        protected void WriteLiteralTo(TextWriter writer, string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                writer.Write(value);
+            }
+        }
+
+        protected string HtmlEncodeAndReplaceLineBreaks(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+
+            // Split on line breaks before passing it through the encoder.
+            return string.Join("<br />" + Environment.NewLine,
+                input.Split(new[] { "\r\n" }, StringSplitOptions.None)
```

Comment:
Hmm, I haven't modified any of this code and just moved it. Ideally I do not want to handle these issues in this PR as I currently have 3 PRs with many changes.